socs/agent/vantage_pro2/vantage_pro2.py

```diff
@@ -0,0 +1,293 @@
+        loop_data['soil_temp2'] = byte_data[22] - 90.0
+        loop_data['soil_temp3'] = byte_data[23] - 90.0
+        loop_data['leaf_temp0'] = byte_data[24] - 90.0
+        loop_data['leaf_temp1'] = byte_data[25] - 90.0
+        loop_data['leaf_temp2'] = byte_data[26] - 90.0
+        loop_data['leaf_temp3'] = byte_data[27] - 90.0
+        loop_data['humidity_outside'] = byte_data[28]
+        loop_data['extra_hum0'] = byte_data[29]
+        loop_data['extra_hum1'] = byte_data[30]
+        loop_data['extra_hum2'] = byte_data[31]
+        loop_data['extra_hum3'] = byte_data[32]
+        loop_data['extra_hum4'] = byte_data[33]
+        loop_data['extra_hum5'] = byte_data[34]
+        loop_data['extra_hum6'] = byte_data[35]
+        loop_data['rain_rate'] = byte_data[36]
+        loop_data['uv'] = byte_data[37]
+        loop_data['solar_rad'] = byte_data[38]
+        loop_data['storm_rain'] = byte_data[39] / 100.0
+        loop_data['storm_start'] = byte_data[40]
+        loop_data['day_rain'] = byte_data[41]
+        loop_data['month_rain'] = byte_data[42]
+        loop_data['year_rain'] = byte_data[43]
+        loop_data['day_ET'] = byte_data[44] / 1000.0
+        loop_data['month_ET'] = byte_data[45] / 100.0
+        loop_data['year_ET'] = byte_data[46] / 100.0
+        loop_data['soil_moisture0'] = byte_data[47]
+        loop_data['soil_moisture1'] = byte_data[48]
+        loop_data['soil_moisture2'] = byte_data[49]
+        loop_data['soil_moisture3'] = byte_data[50]
+        loop_data['leaf_wetness0'] = byte_data[51]
+        loop_data['leaf_wetness1'] = byte_data[52]
+        loop_data['leaf_wetness2'] = byte_data[53]
+        loop_data['leaf_wetness3'] = byte_data[54]
+        loop_data['inside_alarm0'] = byte_data[55]
+        loop_data['inside_alarm1'] = byte_data[56]
+        loop_data['rain_alarm'] = byte_data[57]
+        loop_data['outside_alarm0'] = byte_data[58]
+        loop_data['outside_alarm1'] = byte_data[59]
+        loop_data['extra_temp_hum_alarm0'] = byte_data[60]
+        loop_data['extra_temp_hum_alarm1'] = byte_data[61]
+        loop_data['extra_temp_hum_alarm2'] = byte_data[62]
+        loop_data['extra_temp_hum_alarm3'] = byte_data[63]
+        loop_data['extra_temp_hum_alarm4'] = byte_data[64]
+        loop_data['extra_temp_hum_alarm5'] = byte_data[65]
+        loop_data['extra_temp_hum_alarm6'] = byte_data[66]
+        loop_data['soil_leaf_alarm0'] = byte_data[67]
+        loop_data['soil_leaf_alarm1'] = byte_data[68]
+        loop_data['soil_leaf_alarm2'] = byte_data[69]
+        loop_data['soil_leaf_alarm3'] = byte_data[70]
+        loop_data['transmitter_battery_status'] = byte_data[71]
+        loop_data['console_battery_voltage'] = ((byte_data[72]*300)/512)/100.0
+        loop_data['forecast_icons'] = byte_data[73]
+        loop_data['forecast_rule_num'] = byte_data[74]
+        loop_data['time_sunrise'] = byte_data[75]
+        loop_data['time_sunset'] = byte_data[76]
+
```

Comment:
This is fine, but you should document what all of these values mean and the units of each. This should probably go in the docstring of the function.